### PR TITLE
feat(nargo): add flag to verify created proofs

### LINKED ISSUE #719: Add nargo command to both prove and verify
# Problem

Proving and immediately verifying afterward is a common usecase but has no single command to do both. It would also be nice if there were an option to do so without writing to the file system between the two steps.

# Solution

Add a `nargo run` command to prove and verify.

# Alternatives considered

# Additional context

### PR DESCRIPTION
**Pull request by TomAFrench**
# Related issue(s)

Resolves #719 

# Description

**Draft while waiting for #725 to be merged**

## Summary of changes

This PR adds a `--checked` flag to `nargo prove` which immediately verifies the proof after generation. Most of the changes in this PR are to change to have I/O being done explicitly rather than implicitly through `execute_program` to avoid having to write to/read from the toml files unnecessarily.

## Dependency additions / changes

N/A

## Test additions / changes

Integration tests now make use of this flag rather than calling `verify_with_path`. 

# Checklist

- [x] I have tested the changes locally.
- [x] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` with default settings.
- [x] I have [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) this PR to the issue(s) that it resolves.
- [x] I have reviewed the changes on GitHub, line by line.
- [x] I have ensured all changes are covered in the description.

# Additional context

### CODE REVIEW

File: crates/nargo/src/cli/mod.rs
Comment (jfecher) [inline crates/nargo/src/cli/mod.rs:85]: There seems to be precedence for the `--checked` name, but wouldn't `--verify` or similar be clearer?

File: crates/nargo/src/cli/prove_cmd.rs
Comment (jfecher) [inline crates/nargo/src/cli/prove_cmd.rs:74]: Nitpick, but lets avoid creating an unnecessary block here. It prevents these new variables from being used elsewhere but I don't think it adds enough to merit introducing this new style to the code base.

**Comment by jfecher** [comment]
Looks good, I'll approve once merge conflicts are fixed

**Review by kevaundray — APPROVED** [review]

**Pull request merged**
